File: 1.x/localsim-web/models/infra/road/node.js
//function Node(x, y, dir){
//    //Signal.call(this);
//    Observable.call(this);
//
//    //this.events = ['move', 'destroy'];
//
//    this.x = x;
//    this.y = y;
//    this.dir = dir;
//
//    this.move = function(x, y){
//        this.x = x;
//        this.y = y;
//        this.fire('move');
//    };
//
//    this.deconstruct = function(){
//
//    };
//}

class Node extends Signal{

    constructor(x, y, dir){
        super();
        this.x = x;
        this.y = y;
        this.dir = dir;
        this.events = ['move', 'destroy'];
    }

    move(x, y){
        let change = false;

        if(x != null && x != this.x){
            this.x = x;
            change = true;
        }

        if(y != null && y != this.x) {
            this.y = y;
            change = true;
        }

        if(change){
            this.fire('move');
        }
    }

    destroy(){
        this.fire('destroy');
    }

    deconstruct() {
        const cls = "localsim.models.infra.road.node.Node";
        const args = [this.x, this.y, this.dir];
        const kwargs = {};
        return [cls, args, kwargs]
    }

}

File: 1.x/localsim/models/infra/road/node.py
from localsim.utils import signal


class Node(signal.Signal):
    events = ['move', 'destroy']

    def __init__(self, x, y, dir_):
        self._x = x
        self._y = y
        self.dir = dir_

    def deconstruct(self):
        fullpath = '.'.join([self.__class__.__module__, self.__class__.__name__])
        args = [self.x, self.y, self.dir]
        kwargs = {}
        return [fullpath, args, kwargs]

    def __del__(self):
        self.fire('destroy')

    def move(self, x=None, y=None):
        change = False

        if x is not None and x != self._x:
            self._x = x
            change = True

        if y is not None and y != self._y:
            self._y = y
            change = True

        if change:
            self.fire('move')

    @property
    def x(self):
        return self._x

    @property
    def y(self):
        return self._y

    def __repr__(self):
        return '<%s: (%s, %s)>' % (self.__class__.__name__, self.x, self.y)
